akd_quorum/src/node/mod.rs

```diff
@@ -0,0 +1,2217 @@
+                }
+                Ok(_received) => {
+                    let _ = sender.send(_received).await.map_err(|_| {
+                        QuorumOperationError::Communication(
+                            crate::comms::CommunicationError::SendError(
+                                "Failed to transmit public message to node processing handler"
+                                    .to_string(),
+                            ),
+                        )
+                    })?;
+                }
+            }
+        }
+    }
+
+    async fn flush_message_queue(&self) {
+        let mut m_queue = {
+            let mut queue_guard = self.state.message_queue.write().await;
+            let out = queue_guard.clone();
+            *queue_guard = vec![];
+            out
+            // drop the lock
+        };
+        // turn it into a queue
+        m_queue.reverse();
+
+        while let Some(msg) = m_queue.pop() {
+            let o_target = match &msg {
+                NodeMessage::Internal(from, _) => Some(*from),
+                _ => None,
+            };
+
+            match (self.handle_message(msg).await, o_target) {
+                (Ok(Some(result)), Some(to)) => match result.serialize() {
+                    Ok(reply) => {
+                        let nonce = self.state.nonce_manager.get_next_outgoing_nonce(to).await;
+                        match self.encrypt_message(to, reply, nonce).await {
+                            Ok(message) => {
+                                if let Err(err) =
+                                    self.comms.send_and_maybe_receive(message, None).await
+                                {
+                                    error!(
+                                        "Failed to send reply for backlogged message\n{}",
+                                        QuorumOperationError::from(err)
+                                    );
+                                }
+                            }
+                            Err(err) => {
+                                error!("Failed to encrypt reply for backlogged message\n{}", err);
+                            }
+                        }
+                    }
+                    Err(err) => {
+                        error!(
+                            "Failed to serialize reply for backlogged message\n{}",
+                            QuorumOperationError::from(err)
+                        );
+                    }
+                },
+                (Ok(_), _) => {
+                    // no-op, message handled with no reply
+                }
+                (Err(err), _) => {
+                    error!("Error handling backlogged message: {}", err);
+                }
+            }
+        }
+    }
+
+    async fn inter_node_message_handler(
+        &self,
+        receiver: &mut tokio::sync::mpsc::Receiver<PublicNodeMessage<H>>,
+    ) -> Result<(), QuorumOperationError> {
+        loop {
+            tokio::select! {
```

Comment:
This macro cancels other threads once first one is completed. I'd guess this may lead to dropped messages or interrupted executions in the cancelled thread. Does node tolerate that?